source/common/tcp_proxy/upstream.h

```diff
@@ -171,11 +171,12 @@ class HttpUpstream : public GenericUpstream, protected Http::StreamCallbacks {
     // Http::ResponseDecoder
     void decode1xxHeaders(Http::ResponseHeaderMapPtr&&) override {}
     void decodeHeaders(Http::ResponseHeaderMapPtr&& headers, bool end_stream) override {
-      if (!parent_.isValidResponse(*headers) || end_stream) {
+      bool isValidResponse = parent_.isValidResponse(*headers);
```

Comment:
can you please rename to is_valid_response?

Reply:
@0xquark

Reply:
Done